clusterloader2/pkg/measurement/common/api_availability_metrics.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"time"
+)
+
+type apiAvailabilityMetrics struct {
+	numSuccesses               int
+	numFailures                int
+	maxConsecutiveFailedProbes int
+	consecutiveFailedProbes    int
+}
+
+type apiAvailabilityOutput struct {
+	ClusterMetrics             bool          `json:"clusterMetrics"`
+	HostIP                     string        `json:"hostIP"`
+	AvailabilityPercentage     float32       `json:"availabilityPercentage"`
+	LongestUnavailableDuration time.Duration `json:"longestUnavailableDuration"`
+}
+
+// number of consecutive failed probes is > 0 signifies previous probe was a failure
+func (a *apiAvailabilityMetrics) previousProbeFailure() bool {
+	return a.consecutiveFailedProbes > 0
+}
+
+// Check if the recently concluded consecutive failed number of probes is
+// higher than the max consecutive failed number of probes so far
+// if yes, then Update max consecutive failed probes
+func (a *apiAvailabilityMetrics) updateMaxConsecutiveFailuresIfNeeded() {
+	if a.consecutiveFailedProbes > a.maxConsecutiveFailedProbes {
+		a.maxConsecutiveFailedProbes = a.consecutiveFailedProbes
+	}
+}
+
+func (a *apiAvailabilityMetrics) updateFailureMetrics() {
+	a.numFailures++
+	a.consecutiveFailedProbes++
+}
+
+func (a *apiAvailabilityMetrics) updateSuccessMetrics() {
+	a.numSuccesses++
+	if a.previousProbeFailure() {
```

Comment:
let's remove this check and do that unconditionally - it's not needed.

if the previous attempt was success, consecutiveFailedProbes=0, but maxing with 0 and setting it to 0 don't hurt